notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/NonValidatingNotaryServerFlowImpl.kt

```diff
@@ -0,0 +1,157 @@
+package com.r3.corda.notary.plugin.nonvalidating
+
+import com.r3.corda.notary.plugin.common.NotarisationRequestImpl
+import com.r3.corda.notary.plugin.common.TransactionParts
+import com.r3.corda.notary.plugin.common.sendUniquenessServiceCommitStatus
+import com.r3.corda.notary.plugin.common.validateRequestSignature
+import net.corda.v5.application.crypto.DigitalSignatureVerificationService
+import net.corda.v5.application.flows.CordaInject
+import net.corda.v5.application.flows.FlowEngine
+import net.corda.v5.application.flows.InitiatedBy
+import net.corda.v5.application.flows.ResponderFlow
+import net.corda.v5.application.membership.MemberLookup
+import net.corda.v5.application.messaging.FlowSession
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.util.debug
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.ledger.common.Party
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+import net.corda.v5.ledger.utxo.uniqueness.client.LedgerUniquenessCheckerClientService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * The server-side implementation of the non-validating notary logic.
+ * This will be initiated by the client side of this notary plugin: [NonValidatingNotaryClientFlowImpl]
+ */
+@InitiatedBy(protocol = "non-validating-notary")
+class NonValidatingNotaryServerFlowImpl : ResponderFlow {
+
+    @CordaInject
+    private lateinit var clientService: LedgerUniquenessCheckerClientService
+
+    @CordaInject
+    private lateinit var serializationService: SerializationService
+
+    @CordaInject
+    private lateinit var signatureVerifier: DigitalSignatureVerificationService
+
+    @CordaInject
+    private lateinit var flowEngine: FlowEngine
+
+    @CordaInject
+    private lateinit var memberLookup: MemberLookup
+
+    private var transactionId: SecureHash? = null
+
+    private companion object {
+        val logger: Logger = LoggerFactory.getLogger(this::class.java)
+    }
+
+    /**
+     * The main logic is implemented in this function.
+     *
+     * The logic is very simple in a few steps:
+     * 1. Receive and unpack payload from client
+     * 2. Run initial validation (signature etc.)
+     * 3. Run verification
+     * 4. Commit to the database using the [LedgerUniquenessCheckerClientService]
```

Comment:
```suggestion
     * 4. Request uniqueness checking using the [LedgerUniquenessCheckerClientService]
```
Committing to the database is an implementation detail of what should be a black box